x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const getFollowPattern = (prefix = '', suffix = '', template = '{{leader_index}}') => (
+  prefix + template + suffix
+);
+
+/**
+ * Generate an array of indices preview that would be generated for an auto-follow pattern.
+ * It concatenates the prefix + the leader index pattern populated with values + the suffix
+ *
+ * Example of the array returned:
+ * ["prefix_leader-index-0_suffix", "prefix_leader-index-1_suffix", "prefix_leader-index-2_suffix"]
+ */
+export const getPreviewIndicesFromAutoFollowPattern = ({
```

Comment:
Can we add a couple simple tests to this function and `getPrefixSuffixFromFollowPattern`?